aQute.libg/src/aQute/lib/zip/ZipUtil.java

```diff
@@ -54,6 +54,7 @@ public static String cleanPath(String path) {
 				int lastSlash = clean.lastIndexOf("/");
 				if (lastSlash == -1) {
 					if (clean.length() == 0) {
+						// bad design, this is a common outcome
```

Comment:
We should not throw exceptions for input validation. Wrong input is part of the contract of this method. Exceptions, at least in bnd, are a failure of the code because the inputs are outside the contract.